src/cli/rustup_mode.rs

```diff
@@ -1021,6 +1023,10 @@ fn update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
         cfg.temp_cfg.clean();
     }
 
+    if !self_update && self_update_mode == "check-only" {
```

Comment:
Should we check even if `--no-self-update` is specified?  I don't mind if the thought is "yes we should" but I wanted to raise it as a potential inconsistency here.

Reply:
I think if it's check-only then it's better to always check for updates. I simplified the logic. In fact the first condition is useless.

Reply:
Also if the user has set check-only I think they may no longer use --no-self-update.

Reply:
Simply put, --no-self-update only useful when auto_self_update is enable and #[cfg(not(feature = "no-self-update"))].

Reply:
Priority: no-self-update feature > self_update_mode > no-self-update args.